openedx/core/djangoapps/util/management/commands/run_aggregator.py

```diff
@@ -0,0 +1,272 @@
+            help='Number of completions to perform.'
+        )
+
+    def handle(self, *args, **options):
+        test = options.get('test')
+
+        if not hasattr(self, test):
+            raise CommandError('%s not found.' % test)
+
+        self.setUp(**options)
+        getattr(self, test)()
+
+    def setUp(self, **options):
+        """ Set up the course and users. """
+
+        self.course_breadth = options['course_breadth']
+        self.learners_count = options['learners']
+        self.completions_count = options['completions']
+
+        self.executed_queries = []
+        self.timer = default_timer
+        self.store = modulestore()
+        self.blocks = []
+
+        with self.store.default_store(ModuleStoreEnum.Type.split):
+            fields = {'display_name': 'Course', 'start': datetime.datetime(2015, 1, 1, 1, tzinfo=pytz.utc)}
+            self.course = modulestore().create_course(
+                'completion',
+                '101',
+                unicode(random.randint(1, 9999)),
+                ModuleStoreEnum.UserID.test,
+                fields=fields
+            )
+
+            with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+                with self.store.bulk_operations(self.course.id):
+                    for __ in range(self.course_breadth[0]):
+                        chapter = self._create_block(parent=self.course, category='chapter')
+                        for __ in range(self.course_breadth[1]):
+                            sequence = self._create_block(parent=chapter, category='sequential')
+                            for __ in range(self.course_breadth[2]):
+                                vertical = self._create_block(parent=sequence, category='vertical')
+                                self.blocks += [
+                                    self._create_block(
+                                        parent=vertical, category='html'
+                                    ) for __ in range(self.course_breadth[3])
+                                ]
+                                self.store.publish(vertical.location, ModuleStoreEnum.UserID.test)
+
+            self.course = self.store.get_course(self.course.id)
+
+        self.users = [UserFactory.create() for __ in range(self.learners_count)]
+        for user in self.users:
+            CourseEnrollmentFactory(user=user, course_id=self.course.id)
+
+    def _create_block(self, parent, category):
+        fields = {'display_name': category, 'start': datetime.datetime(2015, 1, 1, 1, tzinfo=pytz.utc)}
+        return self.store.create_child(ModuleStoreEnum.UserID.test, parent.location, category, fields=fields)
+
+    def _print_results_header(self, test_name, time_taken=None):
+        """ Print header. """
+        print u"\n"
+        print u"----- Completion Aggregator Performance Test Results -----"
+        print u"Test: {}".format(test_name)
+        print u"Course: {}".format(self.course.id)
```

Comment:
If you plan to move this to completion_aggregator, use python3 compatible syntax:

```python
from __future__ import absolute_import, division, print_function, unicode_literals
print("\n")
```

Reply:
Done.